### PR TITLE
Preload material icons, allow to preload other fonts

### PR DESCRIPTION
**Pull request by Oksydan**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | CLS in header caused by material icons can be reduced by preloading material icons font. Unfortunately that icon font with version > 6.X got some weight. It is 125 KB for .woff2 format which is rly high. Comparing it to version that is classic theme using it's twice as heavy. I was thinking about preloading font via query param added to font-face src but it is impossible via `webpack-font-preload-plugin` plugin since it is including files query params with extension. It causing to fail at isFont condition. We can think about forking it or create our own plugin. 
| Type?             | improvement
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes https://github.com/PrestaShop/hummingbird/issues/252 - partially 
| How to test?      | Build theme and check waterfall in network tab and loading of fonts in FO.
| Possible impacts? | no

Comparison - throttling 3G fast:

### Before:

https://user-images.githubusercontent.com/25306716/211175116-77de21e3-9e3b-446c-a00e-40731ae744aa.mov

### After:

https://user-images.githubusercontent.com/25306716/211175123-10a10083-1353-4ee3-b193-d39f03351a59.mov

### CODE REVIEW

**Review by NeOMakinG — APPROVED** [review]
That's cool we needed it, forking is most of the time increasing the maintenance cost more than anything

Anyway, even my comment is nitpicking so it's a 🍬 for me!

File: templates/_partials/preload.tpl [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{$themeDir = _PS_THEME_DIR_}
+{$preloadFilePath = "`$themeDir`assets/preload.html"}
+{$assetsUrl = $urls.theme_assets}
+
+{if file_exists($preloadFilePath)}
+  {capture name="preloadBlock"}{include file=$preloadFilePath}{/capture}
+  {$replaceString = "href=\"$assetsUrl"}
+
+  {$smarty.capture.preloadBlock|replace:'href="../':"href=\"$assetsUrl" nofilter}
+{/if}
```
[the remark this answers is not in the document]
Comment (Oksydan) [reply templates/_partials/preload.tpl:10]: Hmm I am not sure how we could achieve it easily, path to file might be different between environments. We can't predict relative path to file or absolute at build time.

**Review by 0x346e3730 — APPROVED** [review]

File: templates/_partials/preload.tpl (hunk 1 above)
[the remark this answers is not in the document]
Comment (Oksydan) [reply templates/_partials/preload.tpl:10]: I didn't notice that I assigned variable there for no reason 🤔. 
I am fixing it

**Review by NeOMakinG — APPROVED** [review]

**Review by kpodemski — CHANGES_REQUESTED** [review]
It's been a while... I wanted to approve it and merge it, but there's a package-lock conflict.

**Comment by Oksydan** [comment]
@kpodemski branch rebased. 
Looks like some dependencies were missing in pacakge-lock.json 🤔

**Review by kpodemski — APPROVED** [review]

**Review by kpodemski — CHANGES_REQUESTED** [review]
I wanted to accept it again, but you added Swiper.js. Not sure what's the reason. I don't see the usage of Swiper in Hummingbird.

**Comment by Oksydan** [comment]
@kpodemski fixed. 
Looks like i did mistake during rebase process. 
Swiper has been removed here https://github.com/PrestaShop/hummingbird/commit/5a654afcf5bc382e9d9f7eb01f58e96047a43492 - it's been here when I was working with this feature.

**Comment by kpodemski** [comment]
@Hlavtox this has to be QA'd by a dev, do you think PS SA devs could help you with that?

**Comment by kpodemski** [comment]
@ga-devfront @GytisZum guys, could you validate this PR? let me know if it looks ok for you

File: webpack/webpack.parts.js [hunk 2]
```diff
@@ -170,6 +172,24 @@ exports.externals = () => ({
...
+exports.preloadFonts = () => ({
+  plugins: [
+    new HtmlWebpackPlugin({
+      filename: 'preload.html',
+      templateContent: '{{{preloadLinks}}}',
+      inject: false,
+    }),
+    new FontPreloadPlugin({
+      index: 'preload.html',
+      extensions: ['woff2'],
```
Comment (ga-devfront) [inline webpack/webpack.parts.js:184]: Why only `woff2` I know `eot` is only for IE and we don't support it but `ttf` is supported by all browser.

**Review by ga-devfront — COMMENTED** [review]
Thank's for your work, I just add one question about the configuration.

**Comment by kpodemski** [comment]
thanks @Oksydan ;-)

**Pull request merged**